src/code/sys_math_atan.c

```diff
@@ -77,24 +77,32 @@ static u16 sATan2Tbl[] = {
     0x1FF6, 0x1FFB, 0x2000,
 };
 
-u16 Math_GetAtan2Tbl(f32 x, f32 y) {
+/**
+ * @param y must be >= 0 and <= x
+ * @param x must be >= 0
+ * @return arctan(y/x) as binang, in [0,0x2000] range
+ */
+u16 Math_GetAtan2Tbl(f32 y, f32 x) {
     u16 ret;
 
-    if (y == 0.0f) {
-        ret = sATan2Tbl[0];
+    if (x == 0.0f) {
+        ret = sAtan2Tbl[0];
     } else {
-        s32 tblIdx = ((x / y) * 1024.0f) + 0.5f;
+        s32 tblIdx = ((y / x) * 1024.0f) + 0.5f;
 
-        if (tblIdx >= ARRAY_COUNT(sATan2Tbl)) {
-            ret = sATan2Tbl[0];
+        if (tblIdx >= ARRAY_COUNT(sAtan2Tbl)) {
+            ret = sAtan2Tbl[0];
         } else {
-            ret = sATan2Tbl[tblIdx];
+            ret = sAtan2Tbl[tblIdx];
         }
     }
 
     return ret;
 }
 
+/**
+ * @return angle from (1,0) to (x,y) as binang, in [-0x8000,0x7FFF] range
+ */
```

Comment:
This might deserve an additional warning to note that the argument order is swapped from how atan2 is typically defined in other languages. Notably, the equivalent function in MM even swaps the argument order back to y,x.

Also, you have the same (1,0) thing.